Issue #104:  Bolts: Per-quest points #104

jahabeebs opened the pull request:
closes #104 

## Description
- [X] Update the existing /bolts endpoint to return this additional data
- [X] Use the fuul endpoint `getUserPointsByConversion`
- [X] Since the SDK cannot be used in a next.js project, you'll need to inspect the sdk package to find the correct endpoint. (I may already be calling it)

I couldn't find an endpoint better than the /payouts endpoint for this so I assume this is the correct endpoint. Their docs don't have anything more up to date and their SDK package is an unreadable dist file

- [X] Recommend using the queries to test things out before updating the actual API
- [X] The final outcome will be that in the app we show the per-quest bolts earned (see image below)

- I added a campaignPoints array to the bolts API response that shows the points per campaign, like this
```json
 "campaignPoints": [
                {
                    "campaign": "Deposit collateral",
                    "points": 818
                },
                {
                    "campaign": "Borrow OD",
                    "points": 700
                }
            ]
        }
```

I hit the https://api.fuul.xyz/api/v1/payouts endpoint to get user's bolts per campaign. Unfortunately, we can't use the group_by filter to group by campaign name, because we can only group by "currency_address" or we get a 400 from the API, so I construct the object manually

here's a full sample response from the /bolts endpoint with the added data:

```json
{
    "success": true,
    "data": {
        "OgNFT": null,
        "OgNFV": false,
        "GenesisNFT": false,
        "galxe": false,
        "zealy": true,
        "fuul": {
            "user": {
                "address": "0xa5a3ee1f3e04bf47d246bd778127a557ed13d87d",
                "points": 1518,
                "rank": 42,
                "attributions": 26,
                "conversions": [
                    {
                        "is_referrer": false,
                        "conversion_id": 4,
                        "conversion_name": "Deposit collateral",
                        "total_amount": "818"
                    },
                    {
                        "is_referrer": false,
                        "conversion_id": 2,
                        "conversion_name": "Borrow OD",
                        "total_amount": "700"
                    }
                ]
            },
            "leaderboard": {
                "users": [
                    {
                        "address": "0x17f6ef86d43b0cedca8a214fa36a961c3c7f5c3e",
                        "points": 2673652,
                        "rank": 1,
                        "attributions": 30
                    },
                    {
                        "address": "0x1df428833f2c9fb1ef098754e5d710432450d706",
                        "points": 641765,
                        "rank": 2,
                        "attributions": 107
                    },
                    {
                        "address": "0x52f1f12f0cc811e2e08315d85a08a5a5693689a2",
                        "points": 512896,
                        "rank": 3,
                        "attributions": 26
                    },
                    {
                        "address": "0x3e893426e65cf198d4df266b77cfa46559c815ee",
                        "points": 226304,
                        "rank": 4,
                        "attributions": 30
                    },
                    {
                        "address": "0x9492510BbCB93B6992d8b7Bb67888558E12DCac4",
                        "points": 165074,
                        "rank": 5,
                        "attributions": 128
                    },
                    {
                        "address": "0x9e07ecd4f5074a2eeac9c42df6508e3ec6373ef3",
                        "points": 99057,
                        "rank": 6,
                        "attributions": 30
                    },
                    {
                        "address": "0xaa15eaa5fac15b47d2b8a695d2f84a44c2b46115",
                        "points": 91315,
                        "rank": 7,
                        "attributions": 56
                    },
                    {
                        "address": "0x223c381a3aae44f7e073e66a8295dce2955e0098",
                        "points": 85824,
                        "rank": 8,
                        "attributions": 30
                    },
                    {
                        "address": "0xe04885c3f1419c6e8495c33bdcf5f8387cd88846",
                        "points": 33629,
                        "rank": 9,
                        "attributions": 57
                    },
                    {
                        "address": "0xc9b7f062a8fd06ca0af3055a82692d6d55f64232",
                        "points": 30385,
                        "rank": 10,
                        "attributions": 65
                    },
                    {
                        "address": "0x247b7e245002e6b525782366dec32446f645320d",
                        "points": 24079,
                        "rank": 11,
                        "attributions": 30
                    },
                    {
                        "address": "0x02077764aa4b88b12a145314d5e8a920c754eae2",
                        "points": 17908,
                        "rank": 12,
                        "attributions": 26
                    },
                    {
                        "address": "0x566ba67fe980ac4626de61c8c9f90ff569f214ae",
                        "points": 17191,
                        "rank": 13,
                        "attributions": 26
                    },
                    {
                        "address": "0xf7b10d603907658f690da534e9b7dbc4dab3e2d6",
                        "points": 16824,
                        "rank": 14,
                        "attributions": 26
                    },
                    {
                        "address": "0x5fdb500ba1d9b6be0c164ce609277cd968c5d45e",
                        "points": 16768,
                        "rank": 15,
                        "attributions": 26
                    },
                    {
                        "address": "0x57ef2b98306d2c98f2fcebbdd05090c75982af72",
                        "points": 16739,
                        "rank": 16,
                        "attributions": 26
                    },
                    {
                        "address": "0x028b2b9adca1bd1263086b0b221713a684968817",
                        "points": 16241,
                        "rank": 17,
                        "attributions": 30
                    },
                    {
                        "address": "0x00002eb28505c1202b8d4beb9f5134f6b67b9d8f",
                        "points": 13671,
                        "rank": 18,
                        "attributions": 30
                    },
                    {
                        "address": "0xa83eac81093fd03170260c733caaf1722be5342c",
                        "points": 12942,
                        "rank": 19,
                        "attributions": 26
                    },
                    {
                        "address": "0x6604B79B688a2E78414f1B91D9a4f2EE4Bd51CF6",
                        "points": 12116,
                        "rank": 20,
                        "attributions": 40
                    },
                    {
                        "address": "0xdd594463a4a809eb8bdb0f15b060aa38573f6a14",
                        "points": 10494,
                        "rank": 21,
                        "attributions": 26
                    },
                    {
                        "address": "0xc44caeb7f0724a156806664d2361fd6f32a2d2c8",
                        "points": 10463,
                        "rank": 22,
                        "attributions": 3
                    },
                    {
                        "address": "0x2408b89c36c6e9020dd747f3b64ee58f5467c4d3",
                        "points": 9768,
                        "rank": 23,
                        "attributions": 30
                    },
                    {
                        "address": "0xb5ded36b90ca73ded492d9bdcefdfb4f737a5e0a",
                        "points": 9125,
                        "rank": 24,
                        "attributions": 26
                    },
                    {
                        "address": "0xf64484970e9fcc46e85ee624db5819bc981cd646",
                        "points": 9025,
                        "rank": 25,
                        "attributions": 26
                    },
                    {
                        "address": "0xbb16140fa27913b255da586bc08c3801e6c19a21",
                        "points": 8859,
                        "rank": 26,
                        "attributions": 26
                    },
                    {
                        "address": "0x3038428b55e18346b43c36acc98ee6ba61b4a6c8",
                        "points": 8686,
                        "rank": 27,
                        "attributions": 26
                    },
                    {
                        "address": "0xfb44496ef7ce7dc94da1ec9f45115c80bba1778d",
                        "points": 8331,
                        "rank": 28,
                        "attributions": 27
                    },
                    {
                        "address": "0x052d62a6479e3c027afff55385f2ba53ffe8ba58",
                        "points": 7074,
                        "rank": 29,
                        "attributions": 30
                    },
                    {
                        "address": "0x95e9a0c113aa9931a4230f91ade08a491d3f8d54",
                        "points": 6434,
                        "rank": 30,
                        "attributions": 4
                    },
                    {
                        "address": "0xbad58e133138549936d2576ebc33251be841d3e9",
                        "points": 6406,
                        "rank": 31,
                        "attributions": 26
                    },
                    {
                        "address": "0x8f9be4310f9abb0e5843cc6363908c9b01dfeb3f",
                        "points": 5718,
                        "rank": 32,
                        "attributions": 26
                    },
                    {
                        "address": "0x187089b33e5812310ed32a57f53b3fad0383a19d",
                        "points": 4719,
                        "rank": 33,
                        "attributions": 26
                    },
                    {
                        "address": "0x43ee79ff632229c903bedbbea51a3c9084d30835",
                        "points": 4695,
                        "rank": 34,
                        "attributions": 4
                    },
                    {
                        "address": "0x839395e20bbb182fa440d08f850e6c7a8f6f0780",
                        "points": 3864,
                        "rank": 35,
                        "attributions": 26
                    },
                    {
                        "address": "0x366bf4c8a1517e2ea6cb5085679742ff92f14b54",
                        "points": 2979,
                        "rank": 36,
                        "attributions": 26
                    },
                    {
                        "address": "0x000000000000000000000000000000000000dead",
                        "points": 2674,
                        "rank": 37,
                        "attributions": 2
                    },
                    {
                        "address": "0x22e86ab483084053562ce713e94431c29d1adb8b",
                        "points": 2262,
                        "rank": 38,
                        "attributions": 26
                    },
                    {
                        "address": "0x140d3f60ac840571a3e08e218b823094d4715564",
                        "points": 2172,
                        "rank": 39,
                        "attributions": 26
                    },
                    {
                        "address": "0xac88c8316c57253a16f482aeb0879020be87db25",
                        "points": 1692,
                        "rank": 40,
                        "attributions": 4
                    },
                    {
                        "address": "0xaEAf20615536F34d95f64FC0fAF71E91Fd8812C6",
                        "points": 1642,
                        "rank": 41,
                        "attributions": 1
                    },
                    {
                        "address": "0xa5a3ee1f3e04bf47d246bd778127a557ed13d87d",
                        "points": 1518,
                        "rank": 42,
                        "attributions": 26
                    },
                    {
                        "address": "0xe4b420f15d6d878dcd0df7120ac0fc1509ee9cab",
                        "points": 1441,
                        "rank": 43,
                        "attributions": 26
                    },
                    {
                        "address": "0xeb47dfd9080a93b9e73b7e0e464f7ffad8f424ef",
                        "points": 1275,
                        "rank": 44,
                        "attributions": 1
                    },
                    {
                        "address": "0xb98dffe8f3c069dc265bd18a0ae9f6a7f6f678ab",
                        "points": 970,
                        "rank": 45,
                        "attributions": 1
                    },
                    {
                        "address": "0x0000000000000000000000000000000000000000",
                        "points": 840,
                        "rank": 46,
                        "attributions": 2
                    },
                    {
                        "address": "0x33f6ba34a34d24885dfe166891a4a8fb83643105",
                        "points": 775,
                        "rank": 47,
                        "attributions": 4
                    },
                    {
                        "address": "0x6276016d1120a6ceb6cab9c23b00dcfb2998cbb0",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x116d761ed35a4675ae77b2bc85739c07bd2b633e",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x69c2d719a75fe2e98f4884bf833a12be55413975",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x118785f9a41501f83e313d18ea9c89ac6f85a004",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xa9e383c2b38cf50323bec927b0a03eb45d71ebd7",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x087bae2b77288f4913afce07eaddaff62b1c52a5",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xa1f6874e70f0e9b7938657d7f007e50cb23d5515",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xe11aeed3d34a71328bd75a593d5d95388a81b3b1",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x643b17380f0dd35f9c02b0f213f9b828900eb0bc",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x28aaa7649afd2004262466755c2691b7f0122d98",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x97dd95e25651a5a48a4091f7522f1ee60656d4fc",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x2ed2d997b96fce1d746ccfc343ba6fff1a8dc4cd",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0xc876e90a2b7c6a18ec6c67a0ecb62ac5b7a98a42",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x6a8428cb82c589c67e0b546cfdc9b926e8f323af",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x7cafe0897ca09dc97623b91dea4d66cc9eb114f5",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x42345fef84d2b10e85c76794bf24174adc0ac1b4",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0x5644b1d8ee3319a6a50240f75aaae5ebab865e0e",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xdad1e096a9e5ee7c0c72ac6d312c3627109d3639",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xf0a3c0317df260408e3e703e09e6f6880a8610ae",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0x3c9daa4bf771543edb199ec426a030deed2f9fa8",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xf2d06c6330845f8c35606b595a8c7ab09563e145",
                        "points": 680,
                        "rank": 68,
                        "attributions": 1
                    },
                    {
                        "address": "0x55187a1165ebb441a1bf227fff1eb0d32a65bc46",
                        "points": 615,
                        "rank": 69,
                        "attributions": 1
                    },
                    {
                        "address": "0x762bbc211990d0a356f35e4d500843f59d223c2e",
                        "points": 615,
                        "rank": 69,
                        "attributions": 1
                    },
                    {
                        "address": "0x5c0e1377edf9a8e4f0430118ae5381024186f12d",
                        "points": 610,
                        "rank": 71,
                        "attributions": 1
                    },
                    {
                        "address": "0x0eceea22ae055aea3482b8df7d36a3587f30f97d",
                        "points": 500,
                        "rank": 72,
                        "attributions": 1
                    },
                    {
                        "address": "0xcf017f65819ce8aa3f980d5a55a913238aab725f",
                        "points": 453,
                        "rank": 73,
                        "attributions": 14
                    },
                    {
                        "address": "0xd88f6e4e70e2951f13de1a1e6dde814c89879abd",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xbD680338B781ccBDAafDf25E902F54B4A4974600",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xabb8fc1e63afa6e24af0b53b2940a67ef4900d9e",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x97d07b19bba91223d0df5ef001c8ecc333af92f8",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x6845bad4930d76a16eafee80d43c4803f85c63aa",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x36f341b88d969fd56e8383a99dd13aae96d9b66b",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xaa034e8e59eac06c35da337d5f793850217c11d3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xca68811204ae583aa3eaa6d39c4334d8ff46adac",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x6cde951b5c32bc04dbf466804e0fc5a9e63c8797",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x470fd974458adb241774bf59fedd74026446d231",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xf3b059c887172f2cc52b5e77ee3b2c8b3a32e6ef",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xa11af5ef0116d99d86a93795cf2177888d3f2988",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x760ffd8ea75bd798b73f01a6808921795711749e",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x765a16ca391a6b9249cfa65bf2d14c38722198e3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd3de8ac3a69b758418130f04ea289ad20ba59d48",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x1823fd0cd641a5aa43f1a096d8c29db311977d09",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd19c611be70eeda28ebec9f318fdc3de2d595592",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xf83c8f5afd3d55aa74cea7edd9fb43dc2e8d2efe",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x29a425878fcbc1ae425f29fc9daa25c7fb14966f",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd1543ae2cc815b510d64a404ec58b3d4179cb2b5",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x650036b83283656bca3a4d11671e3dd4229b5720",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x0e5c31cf5eb4a1614a02bebce8ef50bf55fb9de3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd76ff76ac8019c99237bde08d7c39dab5481bed2",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xc5e9d07295a3c8371a5f4e90c9fb49f60fcf2f29",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0x269ef8f0a80ac256f17ab50f60ca0ec5125bdb91",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xade138add5f52c2636950e515559b595a2687e59",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xbF8DDDB85a055133bbF3b52C60efE5b65f4eA074",
                        "points": 400,
                        "rank": 100,
                        "attributions": 1
                    }
                ],
                "totalUsers": 1662
            },
            "campaignPoints": [
                {
                    "campaign": "Deposit collateral",
                    "points": 818
                },
                {
                    "campaign": "Borrow OD",
                    "points": 700
                }
            ]
        }
    }
}
```

pi0neerpat commented on lib/bolts.js:
```diff
@@ -90,11 +94,39 @@ const getSocialPoints = async (address) => {
     }
+}
+
+const getUserPointsByCampaign = async (address) => {
```
This should be in the fuul file, not here.

jahabeebs replied:
I ended up deleting this since it was an unnecessary call and not used elsewhere

pi0neerpat reviewed (changes requested):
- I noticed this data is already being fetched. It looks to be the same data as the "conversions" shown in your example output posted here. Please use this data instead of making an addition request for the same data. 
- Preserve `conversion_id` and `is_referrer` since this will be helpful in the frontend
- Currently all 1600 users are fetched for the leaderboad. This is a good opportunity to upgrade the function to limit the number of items returned. Please change `fetchFuulPagination` to accept a limit argument, and update the usage in `getLeaderboard` to only fetch the first 100 users

jahabeebs commented:
@pi0neerpat 

- Now we re-use the conversion data rather than doing an extra call
- Now we preserve conversion_id and is_referrer
- Changed fetchFuulPagination to accept a limit argument, and updated the usage in getLeaderboard to only fetch the first 100 users

example response:
```json
{
    "success": true,
    "data": {
        "OgNFT": null,
        "OgNFV": true,
        "GenesisNFT": true,
        "galxe": false,
        "zealy": true,
        "fuul": {
            "user": {
                "address": "0xaa15eaa5fac15b47d2b8a695d2f84a44c2b46115",
                "points": 91437,
                "rank": 7,
                "attributions": 57,
                "conversions": [
                    {
                        "is_referrer": false,
                        "conversion_id": 3,
                        "conversion_name": "Deposit collateral using Genesis NFV",
                        "total_amount": "78534"
                    },
                    {
                        "is_referrer": false,
                        "conversion_id": 1,
                        "conversion_name": "Borrow OD using Genesis NFV",
                        "total_amount": "12310"
                    },
                    {
                        "is_referrer": false,
                        "conversion_id": 7,
                        "conversion_name": "Provide liquidity on OD/ETH pool on Camelot",
                        "total_amount": "593"
                    }
                ]
            },
            "leaderboard": {
                "users": [
                    {
                        "address": "0x17f6ef86d43b0cedca8a214fa36a961c3c7f5c3e",
                        "points": 2903722,
                        "rank": 1,
                        "attributions": 31
                    },
                    {
                        "address": "0x1df428833f2c9fb1ef098754e5d710432450d706",
                        "points": 656283,
                        "rank": 2,
                        "attributions": 108
                    },
                    {
                        "address": "0x52f1f12f0cc811e2e08315d85a08a5a5693689a2",
                        "points": 512896,
                        "rank": 3,
                        "attributions": 26
                    },
                    {
                        "address": "0x3e893426e65cf198d4df266b77cfa46559c815ee",
                        "points": 238833,
                        "rank": 4,
                        "attributions": 31
                    },
                    {
                        "address": "0x9492510BbCB93B6992d8b7Bb67888558E12DCac4",
                        "points": 189505,
                        "rank": 5,
                        "attributions": 130
                    },
                    {
                        "address": "0x9e07ecd4f5074a2eeac9c42df6508e3ec6373ef3",
                        "points": 116052,
                        "rank": 6,
                        "attributions": 31
                    },
                    {
                        "address": "0xaa15eaa5fac15b47d2b8a695d2f84a44c2b46115",
                        "points": 91437,
                        "rank": 7,
                        "attributions": 57
                    },
                    {
                        "address": "0x223c381a3aae44f7e073e66a8295dce2955e0098",
                        "points": 87586,
                        "rank": 8,
                        "attributions": 31
                    },
                    {
                        "address": "0xe04885c3f1419c6e8495c33bdcf5f8387cd88846",
                        "points": 37843,
                        "rank": 9,
                        "attributions": 58
                    },
                    {
                        "address": "0xc9b7f062a8fd06ca0af3055a82692d6d55f64232",
                        "points": 30385,
                        "rank": 10,
                        "attributions": 65
                    },
                    {
                        "address": "0x247b7e245002e6b525782366dec32446f645320d",
                        "points": 26677,
                        "rank": 11,
                        "attributions": 31
                    },
                    {
                        "address": "0x028b2b9adca1bd1263086b0b221713a684968817",
                        "points": 17952,
                        "rank": 12,
                        "attributions": 31
                    },
                    {
                        "address": "0x02077764aa4b88b12a145314d5e8a920c754eae2",
                        "points": 17908,
                        "rank": 13,
                        "attributions": 26
                    },
                    {
                        "address": "0x566ba67fe980ac4626de61c8c9f90ff569f214ae",
                        "points": 17191,
                        "rank": 14,
                        "attributions": 26
                    },
                    {
                        "address": "0xf7b10d603907658f690da534e9b7dbc4dab3e2d6",
                        "points": 16824,
                        "rank": 15,
                        "attributions": 26
                    },
                    {
                        "address": "0x5fdb500ba1d9b6be0c164ce609277cd968c5d45e",
                        "points": 16768,
                        "rank": 16,
                        "attributions": 26
                    },
                    {
                        "address": "0x57ef2b98306d2c98f2fcebbdd05090c75982af72",
                        "points": 16739,
                        "rank": 17,
                        "attributions": 26
                    },
                    {
                        "address": "0x00002eb28505c1202b8d4beb9f5134f6b67b9d8f",
                        "points": 15256,
                        "rank": 18,
                        "attributions": 31
                    },
                    {
                        "address": "0x6604B79B688a2E78414f1B91D9a4f2EE4Bd51CF6",
                        "points": 13323,
                        "rank": 19,
                        "attributions": 41
                    },
                    {
                        "address": "0xa83eac81093fd03170260c733caaf1722be5342c",
                        "points": 12942,
                        "rank": 20,
                        "attributions": 26
                    },
                    {
                        "address": "0x2408b89c36c6e9020dd747f3b64ee58f5467c4d3",
                        "points": 10533,
                        "rank": 21,
                        "attributions": 31
                    },
                    {
                        "address": "0xdd594463a4a809eb8bdb0f15b060aa38573f6a14",
                        "points": 10494,
                        "rank": 22,
                        "attributions": 26
                    },
                    {
                        "address": "0xc44caeb7f0724a156806664d2361fd6f32a2d2c8",
                        "points": 10463,
                        "rank": 23,
                        "attributions": 3
                    },
                    {
                        "address": "0xaEAf20615536F34d95f64FC0fAF71E91Fd8812C6",
                        "points": 9918,
                        "rank": 24,
                        "attributions": 2
                    },
                    {
                        "address": "0xb5ded36b90ca73ded492d9bdcefdfb4f737a5e0a",
                        "points": 9125,
                        "rank": 25,
                        "attributions": 26
                    },
                    {
                        "address": "0xf64484970e9fcc46e85ee624db5819bc981cd646",
                        "points": 9025,
                        "rank": 26,
                        "attributions": 26
                    },
                    {
                        "address": "0xbb16140fa27913b255da586bc08c3801e6c19a21",
                        "points": 8859,
                        "rank": 27,
                        "attributions": 26
                    },
                    {
                        "address": "0x3038428b55e18346b43c36acc98ee6ba61b4a6c8",
                        "points": 8686,
                        "rank": 28,
                        "attributions": 26
                    },
                    {
                        "address": "0x052d62a6479e3c027afff55385f2ba53ffe8ba58",
                        "points": 8386,
                        "rank": 29,
                        "attributions": 31
                    },
                    {
                        "address": "0xfb44496ef7ce7dc94da1ec9f45115c80bba1778d",
                        "points": 8331,
                        "rank": 30,
                        "attributions": 27
                    },
                    {
                        "address": "0x95e9a0c113aa9931a4230f91ade08a491d3f8d54",
                        "points": 8017,
                        "rank": 31,
                        "attributions": 5
                    },
                    {
                        "address": "0xbad58e133138549936d2576ebc33251be841d3e9",
                        "points": 6406,
                        "rank": 32,
                        "attributions": 26
                    },
                    {
                        "address": "0x43ee79ff632229c903bedbbea51a3c9084d30835",
                        "points": 5850,
                        "rank": 33,
                        "attributions": 5
                    },
                    {
                        "address": "0x8f9be4310f9abb0e5843cc6363908c9b01dfeb3f",
                        "points": 5718,
                        "rank": 34,
                        "attributions": 26
                    },
                    {
                        "address": "0x187089b33e5812310ed32a57f53b3fad0383a19d",
                        "points": 4719,
                        "rank": 35,
                        "attributions": 26
                    },
                    {
                        "address": "0x839395e20bbb182fa440d08f850e6c7a8f6f0780",
                        "points": 3864,
                        "rank": 36,
                        "attributions": 26
                    },
                    {
                        "address": "0x366bf4c8a1517e2ea6cb5085679742ff92f14b54",
                        "points": 2979,
                        "rank": 37,
                        "attributions": 26
                    },
                    {
                        "address": "0x000000000000000000000000000000000000dead",
                        "points": 2674,
                        "rank": 38,
                        "attributions": 2
                    },
                    {
                        "address": "0x22e86ab483084053562ce713e94431c29d1adb8b",
                        "points": 2262,
                        "rank": 39,
                        "attributions": 26
                    },
                    {
                        "address": "0x140d3f60ac840571a3e08e218b823094d4715564",
                        "points": 2172,
                        "rank": 40,
                        "attributions": 26
                    },
                    {
                        "address": "0xac88c8316c57253a16f482aeb0879020be87db25",
                        "points": 2108,
                        "rank": 41,
                        "attributions": 5
                    },
                    {
                        "address": "0xa5a3ee1f3e04bf47d246bd778127a557ed13d87d",
                        "points": 1518,
                        "rank": 42,
                        "attributions": 26
                    },
                    {
                        "address": "0xe4b420f15d6d878dcd0df7120ac0fc1509ee9cab",
                        "points": 1441,
                        "rank": 43,
                        "attributions": 26
                    },
                    {
                        "address": "0xeb47dfd9080a93b9e73b7e0e464f7ffad8f424ef",
                        "points": 1275,
                        "rank": 44,
                        "attributions": 1
                    },
                    {
                        "address": "0xb98dffe8f3c069dc265bd18a0ae9f6a7f6f678ab",
                        "points": 970,
                        "rank": 45,
                        "attributions": 1
                    },
                    {
                        "address": "0x33f6ba34a34d24885dfe166891a4a8fb83643105",
                        "points": 959,
                        "rank": 46,
                        "attributions": 5
                    },
                    {
                        "address": "0x0000000000000000000000000000000000000000",
                        "points": 840,
                        "rank": 47,
                        "attributions": 2
                    },
                    {
                        "address": "0x6276016d1120a6ceb6cab9c23b00dcfb2998cbb0",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x116d761ed35a4675ae77b2bc85739c07bd2b633e",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x69c2d719a75fe2e98f4884bf833a12be55413975",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x118785f9a41501f83e313d18ea9c89ac6f85a004",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xa9e383c2b38cf50323bec927b0a03eb45d71ebd7",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0x087bae2b77288f4913afce07eaddaff62b1c52a5",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xa1f6874e70f0e9b7938657d7f007e50cb23d5515",
                        "points": 715,
                        "rank": 48,
                        "attributions": 1
                    },
                    {
                        "address": "0xe11aeed3d34a71328bd75a593d5d95388a81b3b1",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x643b17380f0dd35f9c02b0f213f9b828900eb0bc",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x28aaa7649afd2004262466755c2691b7f0122d98",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x97dd95e25651a5a48a4091f7522f1ee60656d4fc",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x2ed2d997b96fce1d746ccfc343ba6fff1a8dc4cd",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0xc876e90a2b7c6a18ec6c67a0ecb62ac5b7a98a42",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x6a8428cb82c589c67e0b546cfdc9b926e8f323af",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x7cafe0897ca09dc97623b91dea4d66cc9eb114f5",
                        "points": 710,
                        "rank": 55,
                        "attributions": 1
                    },
                    {
                        "address": "0x42345fef84d2b10e85c76794bf24174adc0ac1b4",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0x5644b1d8ee3319a6a50240f75aaae5ebab865e0e",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xdad1e096a9e5ee7c0c72ac6d312c3627109d3639",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xf0a3c0317df260408e3e703e09e6f6880a8610ae",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0x3c9daa4bf771543edb199ec426a030deed2f9fa8",
                        "points": 705,
                        "rank": 63,
                        "attributions": 1
                    },
                    {
                        "address": "0xf2d06c6330845f8c35606b595a8c7ab09563e145",
                        "points": 680,
                        "rank": 68,
                        "attributions": 1
                    },
                    {
                        "address": "0x55187a1165ebb441a1bf227fff1eb0d32a65bc46",
                        "points": 615,
                        "rank": 69,
                        "attributions": 1
                    },
                    {
                        "address": "0x762bbc211990d0a356f35e4d500843f59d223c2e",
                        "points": 615,
                        "rank": 69,
                        "attributions": 1
                    },
                    {
                        "address": "0x5c0e1377edf9a8e4f0430118ae5381024186f12d",
                        "points": 610,
                        "rank": 71,
                        "attributions": 1
                    },
                    {
                        "address": "0x0eceea22ae055aea3482b8df7d36a3587f30f97d",
                        "points": 500,
                        "rank": 72,
                        "attributions": 1
                    },
                    {
                        "address": "0xcf017f65819ce8aa3f980d5a55a913238aab725f",
                        "points": 453,
                        "rank": 73,
                        "attributions": 14
                    },
                    {
                        "address": "0xd88f6e4e70e2951f13de1a1e6dde814c89879abd",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xbD680338B781ccBDAafDf25E902F54B4A4974600",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xabb8fc1e63afa6e24af0b53b2940a67ef4900d9e",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x97d07b19bba91223d0df5ef001c8ecc333af92f8",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x6845bad4930d76a16eafee80d43c4803f85c63aa",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x36f341b88d969fd56e8383a99dd13aae96d9b66b",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xaa034e8e59eac06c35da337d5f793850217c11d3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xca68811204ae583aa3eaa6d39c4334d8ff46adac",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x6cde951b5c32bc04dbf466804e0fc5a9e63c8797",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x470fd974458adb241774bf59fedd74026446d231",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xf3b059c887172f2cc52b5e77ee3b2c8b3a32e6ef",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xa11af5ef0116d99d86a93795cf2177888d3f2988",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x760ffd8ea75bd798b73f01a6808921795711749e",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x765a16ca391a6b9249cfa65bf2d14c38722198e3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd3de8ac3a69b758418130f04ea289ad20ba59d48",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x1823fd0cd641a5aa43f1a096d8c29db311977d09",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd19c611be70eeda28ebec9f318fdc3de2d595592",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xf83c8f5afd3d55aa74cea7edd9fb43dc2e8d2efe",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x29a425878fcbc1ae425f29fc9daa25c7fb14966f",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd1543ae2cc815b510d64a404ec58b3d4179cb2b5",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x650036b83283656bca3a4d11671e3dd4229b5720",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0x0e5c31cf5eb4a1614a02bebce8ef50bf55fb9de3",
                        "points": 440,
                        "rank": 74,
                        "attributions": 1
                    },
                    {
                        "address": "0xd76ff76ac8019c99237bde08d7c39dab5481bed2",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xc5e9d07295a3c8371a5f4e90c9fb49f60fcf2f29",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0x269ef8f0a80ac256f17ab50f60ca0ec5125bdb91",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xade138add5f52c2636950e515559b595a2687e59",
                        "points": 405,
                        "rank": 96,
                        "attributions": 1
                    },
                    {
                        "address": "0xbF8DDDB85a055133bbF3b52C60efE5b65f4eA074",
                        "points": 400,
                        "rank": 100,
                        "attributions": 1
                    }
                ],
                "totalUsers": 100
            },
            "campaignPoints": [
                {
                    "campaign": "Deposit collateral using Genesis NFV",
                    "points": 78534
                },
                {
                    "campaign": "Borrow OD using Genesis NFV",
                    "points": 12310
                },
                {
                    "campaign": "Provide liquidity on OD/ETH pool on Camelot",
                    "points": 593
                }
            ]
        }
    }
}
```

pi0neerpat reviewed (approved):
Awesome! So much faster now
